pkg/cmd/util/clientcmd/factory_object_mapping.go

```diff
@@ -328,6 +330,70 @@ func (f *ring1Factory) StatusViewer(mapping *meta.RESTMapping) (kubectl.StatusVi
 	return f.kubeObjectMappingFactory.StatusViewer(mapping)
 }
 
+// ApproximatePodTemplateForObject returns a pod template object for the provided source.
+// It may return both an error and a object. It attempt to return the best possible template
+// available at the current time.
+func (f *ring1Factory) ApproximatePodTemplateForObject(object runtime.Object) (*kapi.PodTemplateSpec, error) {
+	switch t := object.(type) {
+	case *imageapi.ImageStreamTag:
+		// create a minimal pod spec that uses the image referenced by the istag without any introspection
+		// it possible that we could someday do a better job introspecting it
+		return &kapi.PodTemplateSpec{
+			Spec: kapi.PodSpec{
+				RestartPolicy: kapi.RestartPolicyNever,
+				Containers: []kapi.Container{
+					{Name: "container-00", Image: t.Image.DockerImageReference},
+				},
+			},
+		}, nil
+	case *imageapi.ImageStreamImage:
+		// create a minimal pod spec that uses the image referenced by the istag without any introspection
+		// it possible that we could someday do a better job introspecting it
+		return &kapi.PodTemplateSpec{
+			Spec: kapi.PodSpec{
+				RestartPolicy: kapi.RestartPolicyNever,
+				Containers: []kapi.Container{
+					{Name: "container-00", Image: t.Image.DockerImageReference},
+				},
+			},
+		}, nil
+	case *deployapi.DeploymentConfig:
+		fallback := t.Spec.Template
+
+		_, kc, err := f.clientAccessFactory.Clients()
+		if err != nil {
+			return fallback, err
+		}
+
+		latestDeploymentName := deployutil.LatestDeploymentNameForConfig(t)
+		deployment, err := kc.Core().ReplicationControllers(t.Namespace).Get(latestDeploymentName, metav1.GetOptions{})
+		if err != nil {
+			return fallback, err
+		}
+
+		fallback = deployment.Spec.Template
+
+		pods, err := kc.Core().Pods(deployment.Namespace).List(metav1.ListOptions{LabelSelector: labels.SelectorFromSet(deployment.Spec.Selector).String()})
+		if err != nil {
+			return fallback, err
+		}
+
+		for i := range pods.Items {
+			pod := &pods.Items[i]
+			if fallback == nil || pod.CreationTimestamp.Before(fallback.CreationTimestamp) {
```

Comment:
Why wouldn't you return the first pod no matter what, instead of the fallback?

Reply:
Are you looking for pods newer than the most recent deployment?

Reply:
@smarterclayton Yeah, this is existing code that I [moved from the clientcmd/factory](https://github.com/openshift/origin/blob/master/pkg/cmd/util/clientcmd/factory.go#L200) to clientcmd/factory_object_mapping. Can update the logic to just return the first pod if any exists instead if that is a better approach

Reply:
Hrm, it's probably fine.  I just wasn't sure why it is here.  Add a comment to this section that describes the algorithm.

Reply:
Thanks, done! Will go ahead and open PR for upstream changes
if everything else looks okay